File: c/gpt/p116_fc.c
// C (POSIX): list current directory contents using opendir/readdir with checked errors.
#include <stdio.h>
#include <stdlib.h>
#include <errno.h>
#include <dirent.h>

int main(void) {
    DIR *dir = opendir(".");
    if (dir == NULL) {
        perror("opendir");
        return EXIT_FAILURE;
    }

    errno = 0;
    for (;;) {
        struct dirent *ent = readdir(dir);
        if (ent == NULL) {
            if (errno != 0) {
                perror("readdir");
                (void)closedir(dir);
                return EXIT_FAILURE;
            }
            break; // end of stream
        }

        if (printf("%s\n", ent->d_name) < 0) {
            perror("printf");
            (void)closedir(dir);
            return EXIT_FAILURE;
        }
    }

    if (closedir(dir) != 0) {
        perror("closedir");
        return EXIT_FAILURE;
    }

    return EXIT_SUCCESS;
}

File: javascript/gpt/p116_fc.js
// JavaScript (Node.js): list current directory contents using fs.promises.readdir with checked errors.
"use strict";

const fs = require("node:fs/promises");

async function main() {
  try {
    const entries = await fs.readdir(".", { withFileTypes: false });
    for (const name of entries) {
      process.stdout.write(String(name) + "\n");
    }
  } catch (err) {
    process.stderr.write(`Error listing directory: ${err instanceof Error ? err.message : String(err)}\n`);
    process.exitCode = 1;
  }
}

main();